### PR TITLE
RUN-4827 - mostly there to parity with v1

### PR DESCRIPTION
**Pull request by tgoc99**
Kept the old readme in case we want to include that somewhere.

### CODE REVIEW

File: src/api/interappbus/channel/channels-docs.ts [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+     * 
+     * Register middleware that fires before the action.
+     * @param {Channel#ChannelProvider~Middleware} middleware - Function to be executed before invoking the action.
+     * @tutorial Channel.middleware
+     */
+    beforeAction() { }
+    /**
+     * 
+     * Register an error handler. This is called before responding on any error.
+     * @param {function} middleware - Function to be executed in case of an error.\
```
Comment (rdepena) [inline src/api/interappbus/channel/channels-docs.ts]: extra \

**Comment by rdepena** [comment]
@tgoc99 can you make sure the jsdoc-template changes are PRd into this repo: https://github.com/openfin/jsdoc-template

**Comment by tgoc99** [comment]
I reverted the changes to js-doc template.  All I had done is comment out the events so the ones we added for Window didn't show up.  But per our discussion earlier this week, we haven't decided how we are going to do events so that can fall to this [JIRA](https://appoji.jira.com/browse/RUN-4828) to figure out

File: docs.README.md [hunk 2]
```diff
@@ -1,82 +1,241 @@
...
+#### Index of API Methods <span id=index></span>
+* {@link Application#addEventListener addEventListener} _Application_
+* {@link ExternalApplication#addEventListener addEventListener} _ExternalApplication_
+* {@link Frame#addEventListener addEventListener} _Frame_
+* {@link System#addEventListener addEventListener} _System_
+* {@link Window#addEventListener addEventListener} _Window_
+* {@link InterApplicationBus#addSubscribeListener addSubscribeListener} _InterApplicationBus_
+* {@link InterApplicationBus#addUnsubscribeListener addUnsubscribeListener} _InterApplicationBus_
+* {@link Window#animate animate} _Window_
+* [Application](Application.html#Application) _(constructor)_
```
Comment (licui3936) [inline docs.README.md]: It's not applicable in v2.  Should remove it.

File: docs.README.md [hunk 3]
```diff
@@ -1,82 +1,241 @@
...
+</html>
 ```
 
-Note that either an address or a runtime object with version are required to connect
+#### Index of API Methods <span id=index></span>
+* {@link Application#addEventListener addEventListener} _Application_
+* {@link ExternalApplication#addEventListener addEventListener} _ExternalApplication_
+* {@link Frame#addEventListener addEventListener} _Frame_
+* {@link System#addEventListener addEventListener} _System_
+* {@link Window#addEventListener addEventListener} _Window_
```
Comment (licui3936) [inline docs.README.md]: All 'addEventListener ' should be removed in v2.

File: docs.README.md [hunk 4]
```diff
@@ -1,82 +1,241 @@
...
+* {@link Frame.wrap wrap} _Frame_
+* {@link Window.wrap wrap} _Window_
+* {@link Application.wrapSync wrapSync} _Application_
+* {@link ExternalApplication.wrapSync wrapSync} _ExternalApplication_
+* {@link Frame.wrapSync wrapSync} _Frame_
+* {@link Window.wrapSync wrapSync} _Window_
+* {@link Clipboard#write write} _Clipboard_
+* {@link Clipboard#writeHtml writeHtml} _Clipboard_
+* {@link Clipboard#writeRtf writeRtf} _Clipboard_
+* {@link Clipboard#writeText writeText} _Clipboard_
```
Comment (licui3936) [inline docs.README.md]: all globalHotkey functions missing

File: docs.README.md [hunk 5]
```diff
@@ -1,82 +1,241 @@
...
+* {@link Frame#getInfo getInfo} _Frame_
+* {@link Window#getInfo getInfo} _Window_
+* {@link System#getLog getLog} _System_
+* {@link System#getLogList getLogList} _System_
+* {@link System#getMachineId} _System_
+* {@link System#getMinLogLevel getMinLogLevel} _System_
+* {@link Application#getManifest getManifest} _Application_
+* {@link System#getMonitorInfo getMonitorInfo} _System_
+* {@link System#getMousePosition getMousePosition} _System_
+* {@link Window#getNativeWindow getNativeWindow} _Window_
```
Comment (licui3936) [inline docs.README.md]: should remove it in v2

File: docs.README.md [hunk 6]
```diff
@@ -1,82 +1,241 @@
...
+* {@link InterApplicationBus#removeUnsubscribeListener removeUnsubscribeListener} _InterApplicationBus_
+* {@link Window#resizeBy resizeBy} _Window_
+* {@link Window#resizeTo resizeTo} _Window_
+* {@link Application#restart restart} _Application_
+* {@link Window#restore restore} _Window_
+* {@link Application#run run} _Application_
+* {@link Application#scheduleRestart scheduleRestart} _Application_
+* {@link InterApplicationBus#send send} _InterApplicationBus_
+* {@link Notification#sendMessage sendMessage} _Notification_
+* {@link Notification#sendMessageToApplication sendMessageToApplication} _Notification_
```
Comment (licui3936) [inline docs.README.md]: should remove it in v2

File: docs.README.md [hunk 7]
```diff
@@ -1,82 +1,241 @@
...
+* {@link Application#registerUser registerUser} _Application_
+* {@link System#releaseExternalProcess releaseExternalProcess} _System_
+* {@link Window#reload reload} _Window_
+* {@link Channel#ChannelClient#remove remove} _ChannelClient_
+* {@link Channel#ChannelProvider#remove remove} _ChannelProvider_
+* {@link Application#removeEventListener removeEventListener} _Application_
+* {@link ExternalApplication#removeEventListener removeEventListener} _ExternalApplication_
+* {@link Frame#removeEventListener removeEventListener} _Frame_
+* {@link System#removeEventListener removeEventListener} _System_
+* {@link Window#removeEventListener removeEventListener} _Window_
```
Comment (licui3936) [inline docs.README.md]: all 'removeEventListner' should be removed in v2.

**Pull request merged**